plugin/src/test/groovy/org/gradle/testretry/FilterFuncTest.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.testretry
+
+
+import spock.lang.Unroll
+
+class FilterFuncTest extends AbstractGeneralPluginFuncTest {
+
+    @Unroll
+    def "can filter by class include pattern (gradle version #gradleVersion)"() {
```

Comment:
I don't think I saw tests for excludes or filtering by annotation classes. I'm sure you're planning on it. Here's a reminder in case you get busy with other stuff and have to come back to this.

Reply:
Indeed. I didn't add them before putting this proposal up, but they are there now.